src/st/st0/37308.c

```diff
@@ -3,7 +3,248 @@
+                prim->x0 = 0x10;
+                prim->y0 = 0xA0;
+                prim->u0 = 0;
+                prim->v0 = 0;
+                prim->u1 = 0xF0;
+                prim->v1 = 0x10;
+                prim->clut = 0x1F8;
+                prim->priority = 0x1FE;
+
+                prim->drawMode = DRAW_HIDE;
+            } else {
+                prim->x0 = prim->x1 = prim->x2 = prim->x3 = 0x80;
+                prim->y0 = prim->y1 = prim->y2 = prim->y3 = 0xA7;
+                prim->r0 = prim->r1 = prim->r2 = prim->r3 = 0;
+                prim->g0 = prim->g1 = prim->g2 = prim->g3 = 0;
+                prim->b0 = prim->b1 = prim->b2 = prim->b3 = 0;
+
+                if (i == 1) {
+                    prim->b0 = prim->b1 = prim->b2 = prim->b3 = 0x80;
+                } else {
+                    prim->g0 = prim->g1 = prim->g2 = prim->g3 = 0x80;
+                }
+                prim->priority = 0x1FD;
+                prim->drawMode = DRAW_TPAGE | DRAW_TRANSP;
+            }
+        }
+
+        self->step++;
+        break;
+
+    case 6:
+        msgLen = 0;
+        isObtainedTextStored = false;
+        vramX = 0;
+        msg = g_api.relicDefs[relicId].name;
+        while (true) {
+            ch = *msg++;
+            if (ch == 0) {
+                if (isObtainedTextStored) {
+                    break;
+                }
+                isObtainedTextStored = true;
+                msg = g_RelicOrbTexts[0];
+            } else {
+                ch = (ch << 8) | *msg++;
+                chPixSrc = g_api_func_80106A28(ch, 1);
+                if (chPixSrc != NULL) {
+                    chPixDst = &D_801C00B0[msgLen * 0x30];
+                    for (i = 0; i < 0x30; i++) {
+                        *chPixDst++ = *chPixSrc++;
+                    }
+                    LoadTPage(&D_801C00B0[msgLen * 0x30], 0, 0, vramX, 0x100,
+                              0xC, 0x10);
+                    vramX += 3;
+                    msgLen++;
+                }
+            }
+        }
+        self->ext.relicOrb.unk7E = msgLen;
+        self->ext.relicOrb.unk7C = 0;
+        self->step++;
+        break;
+    case 7:
+        // Animates the blue/green rectangle for the Obtain text bg
+        prim = &g_PrimBuf[self->primIndex];
+        prim = prim->next;
+        for (i = 0; prim != NULL; prim = prim->next, i++) {
+            if (i == 0) {
+                prim->x2 -= 3;
+                prim->x3 += 3;
+                prim->y0 = prim->y1 -= 4;
+                prim->y2 = prim->y3 += 2;
+            } else {
+                prim->x0 -= 3;
+                prim->x1 += 3;
+                prim->y0 = prim->y1 -= 2;
+                prim->y2 = prim->y3 += 4;
+            }
+        }
+
+        if (++self->ext.relicOrb.unk7C == 8) {
+            self->ext.relicOrb.unk7C = 0;
+            self->step++;
+        }
+        break;
+
+    case 8:
+        isObtainedTextStored = self->ext.relicOrb.unk7C;
```

Comment:
I kind of think this should be renamed temp or something since it's reused but it's fine either way